Assets/GLTF/Scripts/GLTFLoader.cs

```diff
@@ -191,29 +198,123 @@ protected virtual GameObject CreateNode(Node node)
 			{
 				foreach (var child in node.Children)
 				{
-					var childObj = CreateNode(child.Value);
+					var childObj = CreateNode(child);
 					childObj.transform.SetParent(nodeObj.transform, false);
 				}
 			}
 
 			return nodeObj;
 		}
 
-		protected virtual void CreateMeshObject(Mesh mesh, Transform parent)
+		/// <summary>
+		/// Generate Animation components from glTF animations, and attach to game objects
+		/// </summary>
+		protected void AddAnimationToScene(Scene scene, GameObject sceneObj, Animation animation)
 		{
-			foreach (var primitive in mesh.Primitives)
+			// create the animation clip that will contain animation data
+			AnimationClip clip = new AnimationClip();
+			clip.name = animation.Name ?? "GLTFAnimation";
+
+			// needed because Animator component is unavailable at runtime
+			clip.legacy = true;
+
+			foreach (var channel in animation.Channels)
+			{
+				AnimationCurve[] curves = channel.AsAnimationCurves();
+
+				// TODO Memoize
+				// Map the target Node ID to a GameObject, get its name, and prepend parents' names to generate a path.
+				GameObject node = _nodeMap[channel.Target.Node.Id];
+				string nodePath = "";
+				do
+				{
+					nodePath = node.name + (nodePath == "" ? nodePath : "/" + nodePath);
+					node = node.transform.parent.gameObject;
+				} while (node != null && node != sceneObj);
```

Comment:
Helper function for this would be :+1:, then we can also remove comment.

Reply:
good idea! I think the Memoizing is done!